uniffi_bindgen/src/bindings/kotlin/templates/ErrorTemplate.kt

```diff
@@ -113,7 +113,7 @@ internal open class {{e.name()}} : RustError() {
             {% for value in e.values() -%}
             {{loop.index}} -> return {{e.name()}}Exception.{{value}}(message) as E
             {% endfor -%}
-            else -> throw RuntimeException("Invalid error received: $code, $message")
+            else -> return InternalException(message) as E
```

Comment:
So, this is an underspecified part of UniFFI. Did we actually catch a panic (which has [a specific error code](https://docs.rs/ffi-support/0.4.3/ffi_support/struct.ErrorCode.html#associatedconstant.PANIC)) or did we read garbage data and try to interpret it as an `ExternError`? And how much does the distinction matter?

Since the Kotlin bindings have other places where they check internal invariants and raise `RuntimeException` if they're violated, what I think I'd like to see here is:

* If `code` is `-1` (the special value for a caught Rust panic) then throw `InternalException`
* If it is any other unexpected value, continue throwing `RuntimeException` as before

Reply:
Hmm good point.. and from there, I can probably create a follow-up to rename the `InternalException` to a `PanicException` or something similar across the bindings.